src/optimizer/plugins/CombineCodingAndQuantityValuesOptimizer.ts

```diff
@@ -72,18 +81,14 @@ export default {
             return;
           }
 
-          const siblingPaths = [
+          const siblings = [
             `${basePath}.system`,
             `${basePath}.display`,
             `${basePath}.unit`,
             `${basePath}.value`
-          ];
-          const siblings = rules.filter(
-            otherRule =>
-              otherRule instanceof ExportableCaretValueRule &&
-              rule.path === otherRule.path &&
-              siblingPaths.includes(otherRule.caretPath)
-          ) as ExportableCaretValueRule[];
+          ]
+            .filter(sibling => ruleMap[rule.path] && ruleMap[rule.path][sibling]?.length)
```

Comment:
I think that
```
ruleMap[rule.path] && ruleMap[rule.path][sibling]?.length
```
can be further simplified to:
```
ruleMap[rule.path]?.[sibling]?.length
```
Of course if you change the value in the sibling map to be a rule instead of an array, you can also remove the `.length` at the end.